src/helpers/defineMessagesAnalyzator.ts

```diff
@@ -0,0 +1,114 @@
+import { Rule } from "eslint";
+import * as ESTree from 'estree';
+import { Dictionary, CallExpressionNode } from "./types";
+
+type TrackedMessage = {
+  isReported: boolean,
+  node: ESTree.Property,
+  context: Rule.RuleContext,
+}
+
+type MessageId = string | number;
+
+export default class DefineMessagesDuplicationAnalyzator {
+  private idTracker: Dictionary<MessageId, TrackedMessage> = {};
+  private context?: Rule.RuleContext;
+
+  public setContext = (context: Rule.RuleContext): void => {
```

Comment:
Who calls this?

Reply:
Each linted file has own its own unique context. So it is called in the begging of linting each file.

Reply:
So, the linter calls this? Is there an interface which has this method that we can implement?

Reply:
In some way. The linter has `create` function, which we are implementing. This function is called on each file scan. And as step of this function I call `duplicationAnalyzer.setContext(context);`

If it confusing, I can pass context in `proceedDefineMessagesFunctionCall` so as I do with node.

Reply:
Please check https://github.com/gopollock/eslint-plugin-formatjs-no-id-duplication/pull/2/commits/472c100b72f0e64eda4c2013b0316f2a9f9db602
It might make more sense